Libraries/vendor/tinycolor/tinycolor.js

```diff
@@ -1,503 +1,341 @@
 /**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
  * @providesModule tinycolor
- * @nolint
+ * @flow
  */
+/* eslint no-bitwise: 0 */
 'use strict';
 
-// TinyColor v1.2.1
-// https://github.com/bgrins/TinyColor
-// Brian Grinstead, MIT License
-
-var trimLeft = /^[\s,#]+/,
-    trimRight = /\s+$/,
-    tinyCounter = 0,
-    mathRound = Math.round,
-    mathMin = Math.min,
-    mathMax = Math.max;
-
-function tinycolor (color, opts) {
-    // If we are called as a function, call using new instead
-    if (!(this instanceof tinycolor)) {
-        return new tinycolor(color, opts);
+function tinycolor(color: string | number): number | null {
+  if (typeof color === 'number') {
+    if (color !== (color >>> 0) || color < 0 || color > 0xffffffff) {
+      return null;
     }
-
-    color = (color) ? color : '';
-    opts = opts || { };
-
-    var rgb = inputToRGB(color);
-    this._r = rgb.r,
-    this._g = rgb.g,
-    this._b = rgb.b,
-    this._a = rgb.a,
-
-    this._ok = rgb.ok;
+    return color;
+  }
+
+  var match;
+
+  // Ordered based on occurences on Facebook codebase
+  if ((match = matchers.hex6.exec(color))) {
+    return 0xff000000 + parseInt(match[1], 16);
+  }
+
+  if (names.hasOwnProperty(color)) {
+    return names[color];
+  }
+
+  if ((match = matchers.rgb.exec(color))) {
+    return (
+      0xff000000 + // a
+      parse255(match[1]) * (1 << 16) + // r
+      parse255(match[2]) * (1 << 8) + // g
+      parse255(match[3]) // b
+    );
+  }
+
+  if ((match = matchers.rgba.exec(color))) {
+    return (
+      parse1(match[4]) * (1 << 24) + // a
+      parse255(match[1]) * (1 << 16) + // r
+      parse255(match[2]) * (1 << 8) + // g
+      parse255(match[3]) // b
+    );
+  }
+
+  if ((match = matchers.hex3.exec(color))) {
+    return (
+      parseInt(
+        'ff' + // a
+        match[1] + match[1] + // r
+        match[2] + match[2] + // g
+        match[3] + match[3], // b
+        16
+      )
+    );
+  }
+
+  if ((match = matchers.hex8.exec(color))) {
+    return parseInt(match[1], 16);
+  }
+
+  if ((match = matchers.hsl.exec(color))) {
+    return (
+      0xff000000 + // a
+      hslToRgb(
+        parse360(match[1]), // h
```

Comment:
Notice parse360, parsePercentage, parse1 and parse255. We don't try to "normalize" units to a 0-1 range as before.